ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/client/Web3JExecutionEngineClient.java

```diff
@@ -127,13 +127,16 @@ private static PowBlock eth1BlockToPowBlock(EthBlock.Block eth1Block) {
   }
 
   private void handleError(Throwable error) {
-    if (isErrored.compareAndSet(false, true)) {
-      EVENT_LOG.executionClientIsOffline(error);
+    final long errorTime = lastError.get();
+    if (errorTime == -1 || System.currentTimeMillis() - errorTime > ERROR_REPEAT_DELAY_MILLIS) {
```

Comment:
Probably worth extracting the `-1` value as a `NO_ERROR_TIME` constant.

And we should also inject a `TimeProvider` instance to get the time from rather than using `System.currentTimeMillis` directly.

Reply:
good point!